app/controllers/api/base_controller.rb

```diff
@@ -104,6 +104,21 @@ def api_request?
 
     protected
 
+    def self.add_scoped_search_description_for(resource)
+      search_fields = resource.scoped_search_definition.fields.map do |k,f|
+        info = { :name => k.to_s }
+        if f.complete_value.is_a?(Hash)
+          info[:values] = f.complete_value.keys
+        else
+          # type is unknown for fields that are delegated to external methods
+          # 'string' is a good guess in such cases
+          info[:type] = f.ext_method.nil? ? f.type.to_s : 'string' rescue ''
```

Comment:
when is the rescue used? and shouldn't it return 'string'?

Reply:
I came across a case when `f.type` was raising exception from somewhere in AR reflections. It appeared with only relation when used with Katello. Since it really isn't a crucial piece of functionality I gave up and added rescue instead of spending hours digging in the code. It's probably some race condition in loading Katello's host extensions.

I decided to return `''` meaning "We don't know what the type is". But I can change it to `'string'` if you find it better.